src/useContentPair.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from "react";
+import { createDecoder, createEncoder } from "@waku/core";
```

Comment:
You are using clear text encoders here (version 0).

It would be good from the get go to be cater for other type of codec, meaning designing this with the fact that one may use other codecs in mind.

Also note that it's only in a public chat scenario that one may want to encode and decode on the same content topic.

In any other case:
- broadcast
- 1:1 messages

Then one would and can only create a Decoder or an Encoder. I am not 100% convinced that coupling them is that useful.

Reply:
I was creating these hooks for `web-chat` specifically since it is a first use case developers think of how to use waku. I will cover other use cases if it will be needed.

I assume this code could be an example of [public key broadcast](https://github.com/waku-org/js-waku-examples/blob/52e3fb4afadafb237d9394158c15e676b33f8360/examples/eth-pm/src/BroadcastPublicKey.tsx#L57)

As for an example for `1:1 messages` I am not sure. 

Could you, please, navigate me here?

Reply:
After discussing it I will split Encoder / Decoder

Reply:
Will add it as followup item